main/src/cgeo/geocaching/maps/mapsforge/v6/RenderThemeSettings.java

```diff
@@ -0,0 +1,160 @@
+package cgeo.geocaching.maps.mapsforge.v6;
+
+import android.app.ActionBar;
+import android.os.Bundle;
+import android.preference.CheckBoxPreference;
+import android.preference.ListPreference;
+import android.preference.PreferenceActivity;
+import android.content.SharedPreferences;
+import android.preference.PreferenceCategory;
+import android.preference.PreferenceManager;
+import android.view.Menu;
+import android.view.MenuItem;
+
+import org.mapsforge.map.rendertheme.XmlRenderThemeStyleLayer;
+import org.mapsforge.map.rendertheme.XmlRenderThemeStyleMenu;
+
+import java.util.Locale;
+import java.util.Map;
+
+import cgeo.geocaching.R;
+import cgeo.geocaching.settings.Settings;
+
+public class RenderThemeSettings extends PreferenceActivity implements SharedPreferences.OnSharedPreferenceChangeListener {
+    public static final String RENDERTHEME_MENU = "renderthememenu";
+
+    ListPreference baseLayerPreference;
+    SharedPreferences prefs;
+    XmlRenderThemeStyleMenu renderthemeOptions;
+    PreferenceCategory renderthemeMenu;
+
+    @Override
+    public boolean onCreateOptionsMenu(final Menu menu) {
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(final MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                finish();
+                return true;
+        }
+        return false;
+    }
+    @Override
+    public void onSharedPreferenceChanged(final SharedPreferences preferences,
+                                          final String key) {
+    if (this.renderthemeOptions != null && this.renderthemeOptions.getId().equals(key)) {
+            createRenderthemeMenu();
+        }
+    }
+
+    @SuppressWarnings("deprecation")
+    @Override
+    protected void onCreate(final Bundle savedInstanceState) {
+        setTheme(Settings.isLightSkin() ? R.style.settings_light : R.style.settings);

```

Comment:
Can this be moved after the `super.onCreate(...)` call?

Reply:
As far as I know it cannot, otherwise it doesn't yield any effect. It took me quite a while to figure that out when we switched to standard preferences.